lib/src/replay/mask/posthog_mask_controller.dart

```diff
@@ -32,15 +31,41 @@ class PostHogMaskController {
       PostHogMaskController._privateConstructor(
           Posthog().config?.sessionReplayConfig);
 
-  Future<List<Rect>?> getCurrentScreenRects() async {
+  /// Extracts a flattened list of [ElementData] objects representing the
+  /// renderable elements in the widget tree.
+  ///
+  /// This method traverses the tree of [ElementData] objects and returns a
+  /// list of elements that have no children or only one child.
+  ///
+  /// The method is designed to extract the elements that are directly
+  /// renderable on the screen.
+  ///
+  /// **Returns:**
+  ///   - `List<ElementData>`: A list of [ElementData] objects representing the
+  ///     renderable elements.
+  ///
+  List<ElementData>? getCurrentWidgetsElements() {
     final BuildContext? context = containerKey.currentContext;
 
     if (context == null) {
+      printIfDebug('Error: containerKey.currentContext is null.');
       return null;
     }
-    final ElementData? widgetElementsTree =
-        _widgetScraper.parseRenderTree(context);
 
-    return widgetElementsTree?.extractRects();
+    try {
+      final ElementData? widgetElementsTree =
+          _widgetScraper.parseRenderTree(context);
```

Comment:
```suggestion
      final widgetElementsTree =
          _widgetScraper.parseRenderTree(context);
```

the type is inferred already